nilearn/input_data/base_masker.py

```diff
@@ -19,61 +20,72 @@
 from .._utils.cache_mixin import CacheMixin, cache
 from .._utils.class_inspect import enclosing_scope_name, get_params
 from .._utils.compat import _basestring, izip
-from nilearn._utils.niimg_conversions import (
-    _iter_check_niimg, _check_same_fov)
-
-
-def filter_and_mask(imgs, mask_img_,
-                    parameters,
-                    memory_level=0,
-                    memory=Memory(cachedir=None),
-                    verbose=0,
-                    confounds=None,
-                    copy=True,
-                    sample_mask=None):
+from nilearn._utils.niimg_conversions import _iter_check_niimg
+
+
+def filter_and_extract(imgs, extraction_function,
+                       smoothing_fwhm, t_r,
+                       standardize, detrend, low_pass, high_pass,
+                       confounds, memory, memory_level,
+                       sessions=None,
+                       target_shape=None,
+                       target_affine=None,
+                       copy=True,
+                       sample_mask=None, verbose=0):
+    """Extract representative time series using given function.
+
+    Parameters
+    ----------
+    imgs: images
+        Images to be masked
+
+    extraction_function: function
+        Function used to extract the time series from 4D data.
+
+    For all other parameters refer to NiftiMasker documentation
+    """
+    # Since the calling class can be any *Nifti*Masker, we look for exact type
+    if verbose > 0:
+        class_name = enclosing_scope_name(stack_level=3)
+
     # If we have a string (filename), we won't need to copy, as
     # there will be no side effect
-
     if isinstance(imgs, _basestring):
         copy = False
 
     if verbose > 0:
-        class_name = enclosing_scope_name(stack_level=2)
-
-    mask_img_ = _utils.check_niimg_3d(mask_img_)
+        print("[%s] Loading data from %s" % (
+            class_name,
+            _utils._repr_niimgs(imgs)[:200]))
```

Comment:
I agree with you on the fact that this is annoying. I think that we should add a shorten=True by default. We could show the long version if verbose is higher than 10 for example.
I don't want to do it here because discussions are needed to solve this problem. If it's OK for you, I'll open an issue and we can solve it in another PR.